### PR TITLE
Model/authentication

### PR DESCRIPTION
**Pull request by aburgosu**

### CODE REVIEW

File: src/main/java/com/foundation/trello/util/Authentication.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    private Authentication() {
+        ReadConfiguration reader = ReadConfiguration.getInstance();
+        String consumerKey = reader.getConsumerKey();
+        String consumerSecret = reader.getConsumerSecret();
+        String accessToken = reader.getAccessToken();
+        String tokenSecret = reader.getTokenSecret();
+        String baseUrl = reader.getUrlBase();
+        request = new RequestSpecBuilder().setAuth(RestAssured.oauth(consumerKey, consumerSecret,
+                accessToken, tokenSecret)).setBaseUri(baseUrl).build();
+
```
Comment (JosueRodriguezGarcia) [inline src/main/java/com/foundation/trello/util/Authentication.java]: delete space, line 29

File: src/test/java/com/foundation/trello/model/ReadConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
+
+/**
+ * This class is used for testing of ReadConfiguration class.
+ */
+public class ReadConfigurationTest {
+
```
Comment (JosueRodriguezGarcia) [inline src/test/java/com/foundation/trello/model/ReadConfigurationTest.java]: delete space, line 18.

File: src/test/java/com/foundation/trello/step/BoardStepDef.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import io.restassured.response.Response;
+
+/**
+ * BoardStepDef class.
+ *
+ * @author Raul Choque
+ * @version 0.0.1
+ */
+public class BoardStepDef {
+
```
Comment (JosueRodriguezGarcia) [inline src/test/java/com/foundation/trello/step/BoardStepDef.java]: delete space, line 15.

File: src/main/java/com/foundation/trello/util/Authentication.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * @author Andres Burgos
+ */
+public final class Authentication {
+
+    private static Authentication oauth;
+    private RequestSpecification request;
+
+    /**
+     * This is constructor that init variables.
```
Comment (MadayAlcala) [inline src/main/java/com/foundation/trello/util/Authentication.java]: Please change 'init' for 'initiates' or 'initializes' (third person).

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * @author Andres Burgos
+ */
+public final class ReadConfiguration {
+    private static ReadConfiguration readConfiguration;
+    private Properties properties;
+    private InputStream inputProperties;
+
+    /**
+     * This is constructor for init variables.
```
Comment (MadayAlcala) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:19]: Please change 'init' for 'initiates' or 'initializes' (third person).

File: src/test/java/com/foundation/trello/step/BoardStepDef.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    private RequestManager request = new RequestManager();
+
+    /**
+     * @param endpoint the endpoint parameter define a input endpoint.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/BoardStepDef.java]: Please change 'define' for 'defines'.

File: src/test/resources/feature/board.feature [hunk 7]
```diff
@@ -0,0 +1,21 @@
+Feature: Acceptance test of "board", CRUD's(Create, Read, Update and Delete)
+
+  Scenario: Read a board by id
+    Given   I set up a GET request to "/boards/9reOdft6" endpoint
+#    When    I send the request
```
Comment (JosueRodriguezGarcia) [inline src/test/resources/feature/board.feature]: Erase all comments,

File: src/test/java/com/foundation/trello/step/BoardStepDef.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    @Given("I set up a GET request to {string} endpoint")
+    public void iSetUpRequestToEndpoint(String endpoint) {
+        Response response = request.getRequest(endpoint);
+        System.out.println(response.body() + "----response");
+    }
+
+    /**
+     *  This method send request.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/BoardStepDef.java]: Change 'send' for 'sends'.

File: src/test/java/com/foundation/trello/step/BoardStepDef.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+     */
+    @When("I send the request")
+    public void iSendTheRequest() {
+        // Write code here that turns the phrase above into concrete actions
+        throw new cucumber.api.PendingException();
+    }
+
+    /**
+     * @param status The status parameter define input status.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/BoardStepDef.java]: Chenge 'define' for 'defines'.

File: src/test/resources/feature/board.feature [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+#
+#  Scenario: Update the name a board
+#    Given   I set up a "PUT" request to "/boards/" endpoint
+#    When    I send the request
+#    Then    I get a "200" status code as response
+#
+#  Scenario: Delete a board by id
+#    Given   I set up a "DELETE" request to "/boards/{boards_id}" endpoint
+#    When    I send the request
+#    Then    I get a "200" status code as response
```
Comment (JosueRodriguezGarcia) [inline src/test/resources/feature/board.feature]: add line later at the end.

File: src/test/java/com/foundation/trello/step/RequestManager.java [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+ * RequestManager class.
+ *
+ * @author Raul Choque
+ * @version 0.0.1
+ */
+public class RequestManager {
+    private RequestSpecification request = Authentication.getInstance().getRequestSpecification();
+
+    /**
+     * @param endpoint the endpoint parameter define a input string.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Change 'define' for 'defines'.

File: src/main/java/com/foundation/trello/util/Authentication.java [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+        String accessToken = reader.getAccessToken();
+        String tokenSecret = reader.getTokenSecret();
+        String baseUrl = reader.getUrlBase();
+        request = new RequestSpecBuilder().setAuth(RestAssured.oauth(consumerKey, consumerSecret,
+                accessToken, tokenSecret)).setBaseUri(baseUrl).build();
+
+    }
+
+    /**
+     * @return an instance.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/Authentication.java:33]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/Authentication.java [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    public static Authentication getInstance() {
+        if (oauth == null) {
+            oauth = new Authentication();
+        }
+        return oauth;
+    }
+
+    /**
+     * @return an request specification.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/Authentication.java:44]: Please add a description of what is they do this method

File: src/test/java/com/foundation/trello/step/RequestManager.java [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+                        when().
+                        get(endpoint).
+                        then().
+                        extract().
+                        response();
+        return response;
+    }
+
+    /**
+     * @param endpoint the endpoint parameter define a input string.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Change 'define' for 'defines'.

File: src/test/java/com/foundation/trello/step/RequestManager.java [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+                        when().
+                        delete(endpoint).
+                        then().
+                        extract().
+                        response();
+        return response;
+    }
+
+    /**
+     * @param endpoint The endpoint parameter define a input string.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Change 'define' for 'defines'.

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    private ReadConfiguration() {
+        properties = readConfigurationFile();
+    }
+
+    /**
+     * @return the values the key.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:26]: Please add a description of what is they do this method

File: src/test/java/com/foundation/trello/step/RequestManager.java [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+     * @param data The data parameter define a input map object.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Change 'define' for 'defines'.

File: src/test/java/com/foundation/trello/step/RequestManager.java [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+                        when().
+                        post(endpoint).
+                        then().
+                        extract().
+                        response();
+        return response;
+    }
+
+    /**
+     * @param endpoint The endpoint parameter define a input string.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Change 'define' for 'defines'.

File: src/test/java/com/foundation/trello/step/RequestManager.java [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+     * @param data The data parameter define a input map object.
```
Comment (MadayAlcala) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Change 'define' for 'defines'.

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+                inputProperties.close();
+            } catch (IOException ex) {
+                ex.printStackTrace();
+            }
+        }
+        return properties;
+    }
+
+    /**
+     * @return an instance.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:45]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public static ReadConfiguration getInstance() {
+        if (readConfiguration == null) {
+            readConfiguration = new ReadConfiguration();
+        }
+        return readConfiguration;
+    }
+
+    /**
+     * @return properties.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:56]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public Properties getProperties() {
+        return properties;
+    }
+    /**
+     * @return consumerKey.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:63]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public String getConsumerKey() {
+        return properties.getProperty("consumerKey");
+    }
+
+    /**
+     * @return consumerSecret.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:71]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public String getConsumerSecret() {
+        return properties.getProperty("consumerSecret");
+    }
+
+    /**
+     * @return accessToken.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:79]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public String getAccessToken() {
+        return properties.getProperty("accessToken");
+    }
+
+    /**
+     * @return tokenSecret.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:87]: Please add a description of what is they do this method

File: src/main/java/com/foundation/trello/util/ReadConfiguration.java [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    public String getTokenSecret() {
+        return properties.getProperty("tokenSecret");
+    }
+
+    /**
+     * @return urlBase.
```
Comment (LimbertVargas) [inline src/main/java/com/foundation/trello/util/ReadConfiguration.java:95]: Please add a description of what is they do this method

File: src/test/java/com/foundation/trello/model/ReadConfigurationTest.java [hunk 27]
```diff
@@ -0,0 +1,139 @@
...
+     */
+    @Test
+    public void putRequestName() {
+        given().
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                param("name", "HelloWorld").log().all().
+                when().
+                put("https://api.trello.com/1/boards/9reOdft6").
+                then().assertThat().statusCode(200).log().all().
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/model/ReadConfigurationTest.java]: maybe, bring down line from ".assertThat()" for maintaining the clean code format.

File: src/test/java/com/foundation/trello/model/ReadConfigurationTest.java [hunk 28]
```diff
@@ -0,0 +1,139 @@
...
+    @Test
+    public void postRequest() {
+        Map<String, Object> mymap = new HashMap<String, Object>();
+        mymap.put("name", "Testeando");
+        given().contentType(JSON).body(mymap).
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                when().
+                post("https://api.trello.com/1/boards/").
+                then().assertThat().statusCode(200);
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/model/ReadConfigurationTest.java]: maybe, bring down line from ".assertThat()" for maintaining the clean code format.

File: src/test/java/com/foundation/trello/model/ReadConfigurationTest.java [hunk 29]
```diff
@@ -0,0 +1,139 @@
...
+     * This test verifies that the DELETE return a state 200.
+     */
+    @Test
+    public void deleteRequest() {
+        given().
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                when().
+                delete("https://api.trello.com/1/boards/jVLf29ab").
+                then().assertThat().statusCode(404).
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/model/ReadConfigurationTest.java]: maybe, bring down line from ".assertThat()" for maintaining the clean code format.

File: src/test/java/com/foundation/trello/model/ReadConfigurationTest.java [hunk 30]
```diff
@@ -0,0 +1,139 @@
...
+    @Test
+    public void postRequestWithQueryParams() {
+        given().
+                auth().
+                oauth(consumerKey, consumerSecret, accessToken, tokenSecret).
+                queryParams("name", "Testeando3").
+                contentType(ContentType.JSON).
+                when().
+                post("https://api.trello.com/1/boards/").
+                then().assertThat().statusCode(200).log().all().
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/model/ReadConfigurationTest.java]: maybe, bring down line from ".assertThat()" for maintaining the clean code format.

File: src/test/java/com/foundation/trello/step/BoardStepDef.java [hunk 31]
```diff
@@ -0,0 +1,44 @@
...
+
+    private RequestManager request = new RequestManager();
+
+    /**
+     * @param endpoint the endpoint parameter define a input endpoint.
+     */
+    @Given("I set up a GET request to {string} endpoint")
+    public void iSetUpRequestToEndpoint(String endpoint) {
+        Response response = request.getRequest(endpoint);
+        System.out.println(response.body() + "----response");
```
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/step/BoardStepDef.java]: please, delete Sout.

File: src/test/java/com/foundation/trello/step/RequestManager.java (hunk 11 above)
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Please add a description of what is they do this method

File: src/test/java/com/foundation/trello/step/RequestManager.java (hunk 14 above)
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Please add a description of what is they do this method

File: src/test/java/com/foundation/trello/step/RequestManager.java (hunk 15 above)
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Please add a description of what is they do this method

File: src/test/java/com/foundation/trello/step/RequestManager.java (hunk 18 above)
Comment (LimbertVargas) [inline src/test/java/com/foundation/trello/step/RequestManager.java]: Please add a description of what is they do this method

**Review by JosueRodriguezGarcia — APPROVED** [review]

**Review by MadayAlcala — APPROVED** [review]

**Review by LimbertVargas — APPROVED** [review]

**Pull request merged**